contracts/cw1-whitelist/src/integration_tests.rs

```diff
@@ -0,0 +1,129 @@
+use crate::msg::{AdminListResponse, ExecuteMsg, InstantiateMsg, QueryMsg};
+use anyhow::{anyhow, Result};
+use assert_matches::assert_matches;
+use cosmwasm_std::testing::{mock_env, MockApi, MockStorage};
+use cosmwasm_std::{to_binary, Addr, CosmosMsg, Empty, QueryRequest, WasmMsg, WasmQuery};
+use cw1::Cw1Contract;
+use cw_multi_test::{App, AppResponse, BankKeeper, Contract, ContractWrapper, Executor};
+use derivative::Derivative;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    static ref OWNER: String = "owner".to_string();
+}
+
+fn mock_app() -> App {
+    let env = mock_env();
+    let api = MockApi::default();
+    let bank = BankKeeper::new();
+    let storage = MockStorage::new();
+
+    App::new(api, env.block, bank, storage)
+}
+
+fn contract_cw1() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(
+        crate::contract::execute,
+        crate::contract::instantiate,
+        crate::contract::query,
+    );
+    Box::new(contract)
+}
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct Suite {
+    /// Application mock
+    #[derivative(Debug = "ignore")]
+    pub app: App,
+    /// cw1 whitelist contract address
+    pub whitelist: Cw1Contract,
+}
+
+impl Suite {
+    pub fn init(mutable: bool) -> Result<Suite> {
+        let mut app = mock_app();
+        let cw1_id = app.store_code(contract_cw1());
+
+        let whitelist = app
+            .instantiate_contract(
+                cw1_id,
+                Addr::unchecked(OWNER.clone()),
+                &InstantiateMsg {
+                    admins: vec![OWNER.clone()],
+                    mutable,
+                },
+                &[],
+                "Whitelist",
+                None,
+            )
+            .unwrap();
+
+        Ok(Suite {
+            app,
+            whitelist: Cw1Contract(whitelist),
+        })
+    }
+
+    pub fn freeze(&mut self, contract_addr: &Addr) -> Result<AppResponse> {
+        let freeze_msg: ExecuteMsg = ExecuteMsg::Freeze {};
+        let execute: ExecuteMsg = ExecuteMsg::Execute {
+            msgs: vec![CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: contract_addr.to_string(),
+                msg: to_binary(&freeze_msg)?,
+                funds: vec![],
+            })],
+        };
+        self.app
+            .execute_contract(
+                Addr::unchecked(OWNER.clone()),
+                self.whitelist.addr(),
+                &execute,
+                &[],
+            )
+            .map_err(|err| anyhow!(err))
+    }
+}
+
+#[test]
+fn execute_freeze() {
```

Comment:
Also missliding name, the clue is not to execute freeze, it is to proxy freeze message. Having this test name, and no single comment on it, I assume flow:
* Have single mutable cw1 contract
* Send `freeze`
* Ensure contract is frozen

My proposal is `proxy_freeze_message`.

Reply:
Agree, done.